public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx

```diff
@@ -0,0 +1,278 @@
+          }
+          isInvalid={!this.isValidNumShards()}
+        />
+        <EuiFormRow fullWidth isInvalid={!this.isValidNumShards()} error={null}>
+          <EuiFieldText
+            fullWidth
+            value={typeof shrink.percentage_of_source_shards === "undefined" ? "" : shrink.percentage_of_source_shards}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => {
+              const percentageOfSourceShards = e.target.value;
+              const shrinkObject = { ...action.action };
+              if (percentageOfSourceShards) shrinkObject.shrink.percentage_of_source_shards = percentageOfSourceShards;
+              else delete shrinkObject.shrink.percentage_of_source_shards;
+              onChangeAction(this.cloneUsingString(shrinkObject));
+            }}
+            data-test-subj="action-render-shrink-percentage-of-source-shards"
+          />
+        </EuiFormRow>
+        <EuiSpacer size="s" />
+        <EuiFormCustomLabel
+          title="Target Index Name Template"
+          helpText={`The mustache template to use to form the name of the output shrunken index.
+          If not provided, a default suffix of "_shrunken" will be appended.`}
+          isInvalid={!this.isValidIndexNameTemplateJson()}
+        />
+        <EuiFormRow fullWidth isInvalid={!this.isValidIndexNameTemplateJson()} error={null} style={{ maxWidth: "100%" }}>
+          <DarkModeConsumer>
+            {(isDarkMode) => (
+              <EuiCodeEditor
+                mode="json"
+                theme={isDarkMode ? "sense-dark" : "github"}
+                width="100%"
+                value={action.action.target_index_name_template_json}
+                onChange={(str) => {
+                  const indexNameTemplateJSON = str;
+                  const shrinkObject = { ...action.action };
+                  if (indexNameTemplateJSON) shrinkObject.target_index_name_template_json = indexNameTemplateJSON;
+                  else delete shrinkObject.target_index_name_template_json;
+                  onChangeAction(this.cloneUsingString(shrinkObject));
+                }}
+                setOptions={{ fontSize: "14px" }}
+                aria-label="Code Editor"
+                height="100px"
+              />
+            )}
+          </DarkModeConsumer>
+        </EuiFormRow>
+        <EuiSpacer size="s" />
+        <EuiFormCustomLabel
+          title="Force unsafe"
+          helpText={`If this is set to 'No' then the shrink action will fail for indices which do not have replica shards.`}
+          isInvalid={false}
+        />
+        <EuiFormRow fullWidth isInvalid={false} error={null} style={{ maxWidth: "100%" }}>
+          <EuiRadioGroup
+            options={radios}
+            idSelected={this.action.force_unsafe_input}
+            onChange={(id) => {
+              const forceUnsafe = id === "yes";
+              const shrinkObject = { ...action.action };
+              if (forceUnsafe) {
```

Comment:
Not critical but may be doing a popup or having some warning when this is set to true stating can potentially result in data loss would be preferred I think

Reply:
Added and demoed in a new screenshot